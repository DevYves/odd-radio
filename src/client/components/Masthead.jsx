import React, { Component } from 'react';

class Masthead extends Component {
  constructor(props){
    super(props);
  }
  render(){
    return(
    <div>
        <div className="container masthead">
          <div className="row station-row border">
            <div className="one-third column center ">
              About
            </div>
            <div className="one-third column center">
              Contact
            </div>
            <div className="one-third column center">
              Team
            </div>
          </div>
          <div className="row border">
            <div className="column">
              <div className="container">
                <p>oddrad.io presents a curated collection of campus and community radio stations 
                  from across Canada. Each station here is an independent, volunteer-driven organization 
                  that focuses on supporting its local community.</p>
                <p>oddrad.io collects these stations web streams in a single location so that you can listen and experience the sounds of different cities across the country.</p>
              </div>
            </div>
          </div>
        </div>
    </div>
    );
  }
}

export default Masthead;