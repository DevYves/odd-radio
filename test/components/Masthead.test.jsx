import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import Masthead from '../../src/client/components/Masthead';

describe('Masthead', () => {
 const props = {
   expanded: 'none',
   clickFunction: jest.fn()
 }

 const subject = <Masthead { ...props }/>;

 it('matches snapshot', () => {
   expect(renderer.create(subject)).toMatchSnapshot();
 });

 describe('onClick on first .one-third class', () => {
   const wrapper = mount(subject);

   it('calls clickFunction', () => {
     wrapper.find('.one-third').at(0).simulate('click');
   });
 });

 describe('onClick on second .one-third class', () => {
   const wrapper = mount(subject);

   it('calls clickFunction', () => {
     wrapper.find('.one-third').at(1).simulate('click');
   });
 });

 describe('onClick on third .one-third class', () => {
   const wrapper = mount(subject);

   it('calls clickFunction', () => {
     wrapper.find('.one-third').at(2).simulate('click');
   });
 });

 describe('when instantiated', () => {
   const wrapper = shallow(subject);

   it('when toggleInfo is called without a target, the state expanded should be none', () => {
     wrapper.instance().toggleInfo('none');

     expect(wrapper.state().expanded).toEqual('none');
   });
 });
});





