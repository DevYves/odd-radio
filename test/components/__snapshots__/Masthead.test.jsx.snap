// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masthead matches snapshot 1`] = `
<div
  className="container masthead"
>
  <div
    className="row border masthead-nav"
  >
    <div
      className="one-third column center"
      onClick={[Function]}
    >
      <h5>
        About
      </h5>
    </div>
    <div
      className="one-third column center"
      onClick={[Function]}
    >
      <h5>
        Contact
      </h5>
    </div>
    <div
      className="one-third column center"
      onClick={[Function]}
    >
      <h5>
        Team
      </h5>
    </div>
  </div>
  <div
    className="row border"
  >
    <div
      className="column"
    />
  </div>
</div>
`;
